### PR TITLE
*:add prometheus  and grafana  guide

### PR DESCRIPTION
**Pull request by nolouch**
PTAL @siddontang @shenli @huachaohuang 

Close https://github.com/pingcap/docs/issues/49  https://github.com/pingcap/docs/issues/48

### CODE REVIEW

File: op-guide/prometheus.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# Use prometheus in TiDB
+
+this page shows you how to deploy prometheus and grafana.
```
Comment (shenli) [inline op-guide/prometheus.md]: this -> This

File: op-guide/prometheus.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+##Preparation
```
Comment (shenli) [inline op-guide/prometheus.md]: ##Preparation -> ## Preparation

File: op-guide/prometheus.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+Before you start, make sure that: 
+
++ Installed the [Pushgateway](https://github.com/prometheus/pushgateway) and [Prometheus](https://prometheus.io/docs/introduction/install)
++ Started the TiDB with [metrics-addr](https://github.com/pingcap/docs/blob/master/op-guide/configuration.md#--metrics-addr)
```
Comment (shenli) [inline op-guide/prometheus.md]: User should also start PD/TiKV with metrics-addr parameter.

File: op-guide/prometheus.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    static_configs:
+      - targets: ['192.168.199.118:9091'] 
+
+```
+
+set `honor_labels:true` if you use pushgateway. and if you use docker start tidb, you should add option `-h $HOSTNAME` to define hostname in docker. After that, you can start prometheus with `prometheus.yml`.
+
+
+## Step 2. To monitor TiDB with Grafana/Prometheus
+Grafana supports querying Prometheus.we use grafana to show result of querying.
```
Comment (shenli) [inline op-guide/prometheus.md]: Prometheus.we -> Prometheus. We

File: op-guide/prometheus.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+Install the [Grafana](http://docs.grafana.org/)
+
+By default, Grafana will be listening on [http://localhost:3000](). The default login is "admin" / "admin".Now,open grafana address in browser.
```
Comment (shenli) [inline op-guide/prometheus.md]: default username/password is

Comment (shenli) [inline op-guide/prometheus.md]: Insert a whitespace before 'Now'

File: op-guide/prometheus.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+###Creating a Prometheus data source
```
Comment (shenli) [inline op-guide/prometheus.md]: Leave a whitespace after '###'. So as the following '###'

File: op-guide/prometheus.md (hunk 1 above)
Comment (siddontang) [inline op-guide/prometheus.md]: Prometheus and Grafana

File: op-guide/prometheus.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+    # metrics_path defaults to '/metrics'
+    # scheme defaults to 'http'.
+    # here target used pushgateway address.
+    static_configs:
+      - targets: ['192.168.199.118:9091'] 
+
+```
+
+set `honor_labels:true` if you use pushgateway. and if you use docker start tidb, you should add option `-h $HOSTNAME` to define hostname in docker. After that, you can start prometheus with `prometheus.yml`.
```
Comment (siddontang) [inline op-guide/prometheus.md]: s/docker/Docker/g

File: op-guide/prometheus.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+Before you start, make sure you have: 
+
++ Installed the [Pushgateway](https://github.com/prometheus/pushgateway) and [Prometheus](https://prometheus.io/docs/introduction/install)
++ Started the TiDB/TiKV/PD with [metrics-addr](https://github.com/pingcap/docs/blob/master/op-guide/configuration.md)
+
+## Step 1. Configure the Prometheus
+
+See the following `prometheus.yml` as an example:
+
+```
```
Comment (siddontang) [inline op-guide/prometheus.md]: add yaml for highlight

File: op-guide/prometheus.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    # scheme defaults to 'http'.
+    # here target used pushgateway address.
+    static_configs:
+      - targets: ['192.168.199.118:9091'] 
+
+```
+
+**Note**:
+ - If you use pushgateway, set `honor_labels:true`.
+ - If you use Docker start TiDB, add the `-h $HOSTNAME` option to define the hostname in Docker. Then, you can start Prometheus using the `prometheus.yml` file.
```
Comment (siddontang) [inline op-guide/prometheus.md]: If we use k8s later, we may not get the hostname easily, so I don't think it is necessary to pass hostname here. 
We can use other values for the `instance` field. 
+ For TiDB, We can use host:port 
+ For PD, we can use PD name or the first advertise-url address 
+ For TiKV, we can use advertise-addr 

/cc @shenli @huachaohuang @overvenus

**Comment by siddontang** [comment]
Rest LGTM

File: op-guide/prometheus.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
+# Use Prometheus and Grafana in TiDB
+
+This page shows you how to deploy Prometheus and Grafana.  
+## Preparation
```
Comment (overvenus) [inline op-guide/prometheus.md]: Insert a blank line above `## Preparation`

**Comment by iamxy** [comment]
LGTM

**Comment by nolouch** [comment]
Ref https://github.com/pingcap/docs/pull/60 maybe i need remove `prometheus.md`   and  keep the grafana json file?
PTAL @siddontang  @shenli

**Comment by siddontang** [comment]
I think we can only keep the json file here now.

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
Do we need add a link for the config?

**Comment by nolouch** [comment]
Maybe we can add links in `monitoring-tidb.md`

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**